plugins/cloudbuild/src/components/Cards/Cards.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useEffect } from 'react';
+import { useWorkflowRuns } from '../useWorkflowRuns';
+import { WorkflowRun, WorkflowRunsTable } from '../WorkflowRunsTable';
+import { Entity } from '@backstage/catalog-model';
+import { WorkflowRunStatus } from '../WorkflowRunStatus';
+import {
+  Link,
+  Theme,
+  makeStyles,
+  LinearProgress,
+  Typography,
+} from '@material-ui/core';
+import {
+  InfoCard,
+  StructuredMetadataTable,
+  errorApiRef,
+  useApi,
+} from '@backstage/core';
+import ExternalLinkIcon from '@material-ui/icons/Launch';
+import { CLOUDBUILD_ANNOTATION } from '../useProjectName';
+
+const useStyles = makeStyles<Theme>({
+  externalLinkIcon: {
+    fontSize: 'inherit',
+    verticalAlign: 'bottom',
+  },
+});
+
+const WidgetContent = ({
+  error,
+  loading,
+  lastRun,
+  branch,
+}: {
+  error?: Error;
+  loading?: boolean;
+  lastRun: WorkflowRun;
+  branch: string;
+}) => {
+  const classes = useStyles();
+  if (error) return <Typography>Couldn't fetch latest {branch} run</Typography>;
```

Comment:
There's a `WarningPanel` component in core that's perfect for error feedback to users